File: update_cv.py
import os, json, re

projects_dir = r"D:\_Portfolio\projects"
cv_path = r"D:\_Company Portfolio\cv.json"

with open(cv_path, 'r', encoding='utf-8') as f:
    cv = json.load(f)

for folder in os.listdir(projects_dir):
    p_path = os.path.join(projects_dir, folder, 'index.html')
    if os.path.exists(p_path):
        with open(p_path, 'r', encoding='utf-8') as f:
            html = f.read()

        dl_match = re.search(r'href="([^"]+)"[^>]*>(?:Play Game|Download)', html)
        vid_match = re.search(r'youtube\.com/embed/([^"?]+)', html)

        for proj in cv.get('projects', []):
            demo = proj.get('demo', '')
            if demo and folder + '/' in demo:
                if vid_match:
                    proj['video'] = 'https://youtu.be/' + vid_match.group(1)
                if dl_match:
                    proj['download'] = dl_match.group(1)

with open(cv_path, 'w', encoding='utf-8') as f:
    json.dump(cv, f, indent=2)

print("Updated OK")

File: update_cv.js
const fs = require('fs');
const path = require('path');

const projectsDir = 'D:/_Portfolio/projects';
let cv = JSON.parse(fs.readFileSync('D:/_Company Portfolio/cv.json', 'utf8'));

fs.readdirSync(projectsDir).forEach(folder => {
  const pPath = path.join(projectsDir, folder, 'index.html');
  if (fs.existsSync(pPath)) {
    const html = fs.readFileSync(pPath, 'utf8');
    
    // Extract Play Game / Download link
    let dlMatch = html.match(/href="([^"]+)"[^>]*>Play Game/);
    if (!dlMatch) dlMatch = html.match(/href="([^"]+)"[^>]*>Download/);
    
    // Extract Video
    let vidMatch = html.match(/youtube\.com\/embed\/([^"?]+)/);
    
    // Find matching project in cv.json by folder name mapping 
    // Find the one where `demo` contains the folder name
    const proj = cv.projects.find(p => p.demo && p.demo.includes(folder + '/'));
    if (proj) {
      if (vidMatch) {
         proj.video = "https://youtu.be/" + vidMatch[1];
      }
      if (dlMatch) {
         proj.download = dlMatch[1];
      }
    }
  }
});

fs.writeFileSync('D:/_Company Portfolio/cv.json', JSON.stringify(cv, null, 2));
console.log("Updated local cv.json with correct links from local project files!");

